src/index.ts

```diff
@@ -325,7 +325,7 @@ You may find the requirements here: ${cyan('https://tauri.app/v1/guides/getting-
     updatePackageJson((pkg) => {
       return {
         ...pkg,
-        name: appName === '.' ? defaults.appName : appName,
+        name: toValidPackageName(appName) === '.' ? defaults.appName : toValidPackageName(appName),
```

Comment:
there should be another function that checks if it is a valid name and it should be used like this
```suggestion
        name: isValidPackageName(appName)? appName: toValidPackageName(appName),
```

and you should do this logic here too https://github.com/tauri-apps/create-tauri-app/blob/e17dc70c319b0558f8be98e738f30ad76a5d2152/src/index.ts#L340-L341

Reply:
Updated.